File: LeetCode/Explore/TopInterview/Amazon/SortingSearching/4_median_of_two_sorted_arrays.cpp
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
      vector<int> vp = nums1;
      for (auto &num : nums2) {
        vp.push_back(num);
      }
      sort(vp.begin(), vp.end());
      if (vp.size() & 1) {
        return (double)vp[vp.size() / 2];
      }
      else {
        return (vp[(vp.size() / 2) - 1] + vp[vp.size() / 2]) * 0.5;
      }
    }
};

File: LeetCode/Explore/TopInterview/Amazon/SortingSearching/4_median_of_two_sorted_arrays.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        vp = nums1 + nums2
        vp.sort()
        
        if len(vp) % 2 == 1:
          return vp[len(vp) // 2] * 1.0
        else:
          return (vp[len(vp) // 2 - 1] + vp[(len(vp) // 2)]) * 0.5
